### PR TITLE
[ticket/14804] Add core event to MCP after merging topics

### PR DESCRIPTION
**Pull request by dsinn**
Checklist:
- [x] Correct branch: master for new features; 3.2.x, 3.1.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master / 3.2.x](https://area51.phpbb.com/docs/master/coding-guidelines.html), [3.1.x](https://area51.phpbb.com/docs/31x/coding-guidelines.html)
- [x] Commit follows commit message [format](https://wiki.phpbb.com/Git#Commit_Messages)

Tracker ticket (set the ticket ID to **your ticket ID**):

https://tracker.phpbb.com/browse/PHPBB3-14804

Renamed $topic_data and defined $to_topic_data in order to avoid ambiguity
and to stop overwriting a variable.

PHPBB3-14804

### CODE REVIEW

File: phpBB/includes/mcp/mcp_forum.php [hunk 1]
```diff
@@ -503,6 +503,20 @@ function merge_topics($forum_id, $topic_ids, $to_topic_id)
 		$redirect = request_var('redirect', "{$phpbb_root_path}viewtopic.$phpEx?f=$to_forum_id&amp;t=$to_topic_id");
 		$redirect = reapply_sid($redirect);
 
+		/**
+		 * Perform additional actions after merging topics.
+		 *
+		 * @event core.mcp_merge_after
+		 * @var	array	all_topic_data			The data from all topics involved in the merge
+		 * @var	int		to_topic_id				The ID of the topic into which the rest are merged
+		 * @since 3.1.10-RC1
```
Comment (rxu) [inline phpBB/includes/mcp/mcp_forum.php]: It's too late for 3.1.10-RC1 as it's already out. Should go to 3.1.11-RC1.

File: phpBB/includes/mcp/mcp_forum.php [hunk 2]
```diff
@@ -503,6 +503,20 @@ function merge_topics($forum_id, $topic_ids, $to_topic_id)
 		$redirect = request_var('redirect', "{$phpbb_root_path}viewtopic.$phpEx?f=$to_forum_id&amp;t=$to_topic_id");
 		$redirect = reapply_sid($redirect);
 
+		/**
+		 * Perform additional actions after merging topics.
+		 *
+		 * @event core.mcp_merge_after
```
Comment (rxu) [inline phpBB/includes/mcp/mcp_forum.php]: The name could be more descriptive, like ```core.mcp_forum_merge_topics_after``` or something like that.

**Comment by dsinn** [comment]
I haven't worked with Travis very much, so I'm not sure if I'm able to manually trigger a build without committing or closing & reopening the PR.

**Comment by marc1706** [comment]
Thanks for the pull request.

**Pull request merged**